### PR TITLE
Bug 1671915 - Change the 'preview' crate to the 'rlb' crate

### PR DESCRIPTION
**Pull request by Dexterp37**
This requires #1264 to be merged first (and to be rebased after that happens)

This lays the foundations to develop the Rust Language Bindings: it renames the 'preview' crate, which is no longer useful, and makes sure we properly update the version numbers when cutting new releases.

### CODE REVIEW

File: docs/dev/cut-a-new-release.md [hunk 1]
```diff
@@ -87,6 +87,8 @@ When CI has finished and is green for your specific release branch, you are read
     cargo publish --verbose
     cd ffi
     cargo publish --verbose
+    cd.. && cd rlb
```
Comment (badboy) [inline docs/dev/cut-a-new-release.md:90]:
```suggestion
    cd .. && cd rlb
```

Gnaaaah, I'm going to automate this away in the coming weeks.

Comment (badboy) [reply docs/dev/cut-a-new-release.md:90]: https://bugzilla.mozilla.org/show_bug.cgi?id=1598291

File: bin/prepare-release.sh [hunk 2]
```diff
@@ -102,21 +102,13 @@ run $SED -i.bak -E \
     "${WORKSPACE_ROOT}/${FILE}"
 run rm "${WORKSPACE_ROOT}/${FILE}.bak"
 
-### GLEAN-PREVIEW ###
+### GLEAN RLB ###
 
 # Update the version of the glean-core dependency
 
-FILE=glean-core/preview/Cargo.toml
```
Comment (badboy) [inline bin/prepare-release.sh:109]: glean-preview was versioned independently of glean-core (it's at 0.0.5).
The code previously here ensured we update the glean-core version it depends on.

This needs changes to update both the glean-core version it depends on and its own version.

File: glean-core/rlb/Cargo.toml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 [package]
-name = "glean-preview"
+name = "glean"
 version = "0.0.5"
```
Comment (badboy) [inline glean-core/rlb/Cargo.toml:3]: As said above, do we immediately bump this to the same version as glean-core (33.y.z)?

File: glean-core/rlb/Cargo.toml [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
 authors = ["Jan-Erik Rediger <jrediger@mozilla.com>", "The Glean Team <glean-team@mozilla.com>"]
-description = "Nice Glean SDK Rust API"
+description = "Glean SDK Rust language bindings"
```
Comment (badboy) [inline glean-core/rlb/Cargo.toml:5]: I can't comment on a moved-only file, but the glean-core/rlb/README.md needs changes too

**Review by badboy — CHANGES_REQUESTED** [review]

File: bin/prepare-release.sh (hunk 2 above)
Comment (Dexterp37) [reply bin/prepare-release.sh:109]: Yes, this is what the file is doing now: i tested it locally and it updates both the glean-core version and its own version.

Comment (badboy) [reply bin/prepare-release.sh:109]: uh fancy, I can now see why. it's both a line with just `version = ...`

File: glean-core/rlb/Cargo.toml (hunk 4 above)
Comment (Dexterp37) [reply glean-core/rlb/Cargo.toml:5]: Done!

**Review by badboy — APPROVED** [review]
r+wc

The `glean-core/rlb/CHANGELOG.md` file can be dropped now. going forward we will document changes in the top-level changelog.

Otherwise good to go.

**Pull request merged**